image_tools/src/showimage.cpp

```diff
@@ -80,73 +86,75 @@ void show_image(
   }
 }
 
-int main(int argc, char * argv[])
+/// Default constructor for no command line args
+/**
+ */
+ShowImage::ShowImage(rclcpp::NodeOptions options)
+: Node("showimage", options)
 {
-  // Pass command line arguments to rclcpp.
-  rclcpp::init(argc, argv);
-
-  // Initialize default demo parameters
-  size_t depth = rmw_qos_profile_default.depth;
-  rmw_qos_reliability_policy_t reliability_policy = rmw_qos_profile_default.reliability;
-  rmw_qos_history_policy_t history_policy = rmw_qos_profile_default.history;
-  bool show_camera = true;
-  std::string topic("image");
-
-  // Force flush of the stdout buffer.
-  // This ensures a correct sync of all prints
-  // even when executed simultaneously within a launch file.
-  setvbuf(stdout, NULL, _IONBF, BUFSIZ);
-
-  // Configure demo parameters with command line options.
-  if (!parse_command_options(
-      argc, argv, &depth, &reliability_policy, &history_policy, &show_camera, nullptr, nullptr,
-      nullptr, nullptr, &topic))
-  {
-    return 0;
+  execute();
+}
+
+/// Constructor for setup with command line args
+ShowImage::ShowImage(rclcpp::NodeOptions options, int argc, char ** argv)
+: Node("showimage", options){
+  if(setup(argc, argv)){
+    execute();
+  }else{
+    rclcpp::shutdown();
   }
+}
 
-  if (show_camera) {
+/// Execute main functions with image subscriber and callback
+/**
+ */
+void ShowImage::execute(){
+  if (show_camera_) {
     // Initialize an OpenCV named window called "showimage".
+      // RCLCPP_INFO(this->get_logger(), "SHOWING THE CAMERA");
+
     cv::namedWindow("showimage", cv::WINDOW_AUTOSIZE);
     cv::waitKey(1);
   }
 
-  // Initialize a ROS node.
-  auto node = rclcpp::Node::make_shared("showimage");
-
-  // Set quality of service profile based on command line options.
   auto qos = rclcpp::QoS(
     rclcpp::QoSInitialization(
-      // The history policy determines how messages are saved until taken by
```

Comment:
I think it's good to leave these comments in since this is code that we expect users to look at.